AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -365,6 +367,15 @@ class RemoteSyncEngine: RemoteSyncEngineBehavior {
         Amplify.Hub.dispatch(to: .dataStore, payload: networkStatusEventPayload)
     }
 
+    func cancelEmitter() {
+        if syncEventEmitter != nil {
+            syncEventEmitter = nil
+        }
+        if readyEventEmitter != nil {
+            readyEventEmitter = nil
+        }
+    }
```

Comment:
Decrement the ARC ([Automatic Reference Count](https://docs.swift.org/swift-book/LanguageGuide/AutomaticReferenceCounting.html)) to 0 so that the instance will be deallocated by the system automatically.

Reply:
Welcome to [memory management in Swift](https://github.com/palpatim/StrongReferenceDemo/blob/master/20160406%20-%20Swift%20Strong%20Reference%20Cycles.key)! :)